src/components/views/rooms/EventTile.tsx

```diff
@@ -1404,20 +1358,48 @@ export class UnwrappedEventTile extends React.Component<EventTileProps, IState>
                         "onMouseEnter": () => this.setState({ hover: true }),
                         "onMouseLeave": () => this.setState({ hover: false }),
                         "onClick": (ev: MouseEvent) => {
-                            dis.dispatch<ShowThreadPayload>({
-                                action: Action.ShowThread,
-                                rootEvent: this.props.mxEvent,
-                                push: true,
-                            });
                             const target = ev.currentTarget as HTMLElement;
-                            const index = Array.from(target.parentElement.children).indexOf(target);
-                            PosthogTrackers.trackInteraction("WebThreadsPanelThreadItem", ev, index);
+                            let index = -1;
+                            if (target.parentElement) index = Array.from(target.parentElement.children).indexOf(target);
+                            switch (this.context.timelineRenderingType) {
+                                case TimelineRenderingType.Notification:
+                                    this.viewInRoom(ev);
+                                    break;
+                                case TimelineRenderingType.ThreadsList:
+                                    dis.dispatch<ShowThreadPayload>({
+                                        action: Action.ShowThread,
+                                        rootEvent: this.props.mxEvent,
+                                        push: true,
+                                    });
+                                    PosthogTrackers.trackInteraction("WebThreadsPanelThreadItem", ev, index ?? -1);
+                                    break;
+                            }
                         },
                     },
                     <>
-                        {sender}
-                        {avatar}
-                        {timestamp}
+                        <div className="mx_EventTile_details">
```

Comment:
I wonder if this possible for this new code to move it to a component. To avoid to have component with thousands of lines it :)

Reply:
Yes definitely does make sense, but this line with the sender is something that is used across all of the other rendering type too, and the refactor is a bit too involved for this task.

Reply:
Fair enough, for the futur :)